nixos/modules/config/xdg/icons.nix

```diff
@@ -19,9 +19,15 @@ with lib;
       "/share/pixmaps"
     ];
 
-    environment.profileRelativeSessionVariables = {
-      XCURSOR_PATH = [ "/share/icons" ];
-    };
+    environment.profileRelativeSessionVariables.XCURSOR_PATH = [
+      "/share/icons"
+      "/share/pixmaps"
+    ];
+
+    environment.sessionVariables.XCURSOR_PATH = [
+      "$HOME/.local/share/icons"
```

Comment:
This is XDG_DATA_HOME, right? The spec doesn't explicitly mention looking there, although I assume that's an omission.

Reply:
Yes this is `XDG_DATA_HOME`
```
$XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.
```

and we don't set `$XDG_DATA_HOME` in nixos.

Reply:
see commit message https://gitlab.freedesktop.org/xorg/lib/libxcursor/commit/2263c196cb0dcb8547b378df7b35f83b8b99c01e

Reply:
Yeah, but `$XDG_DATA_HOME` does not have to be set. In that case it should fallback to `$HOME/.local/share` (it is described in [`xdg-basedir-spec`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html). Unfortunately, we cannot express that in PAM configuration – we can [only](https://github.com/NixOS/nixpkgs/blob/ab9f0b7f06a3e3497a93405e8680209291d2d22d/nixos/modules/config/system-environment.nix#L13) do `$HOME` and `$USER` variables; and `~` prefix which will be processed by Xcursor itself.

The optimal solution would be patching [Xcursor itself](https://gitlab.freedesktop.org/xorg/lib/libxcursor/blob/b84b5d100f193fda0630c4d6fa889cd3e05ca033/src/library.c#L31-47) to recognize basedir-spec, but yuck raw C.

Reply:
How could patching xcursor work? You'd have to give xcursor `config.environment.profiles` and changing them would cause a mass rebuild.

Reply:
It would have to be patched to support xdg_data_dirs.

Reply:
Right, but at the point, it would make a good patch upstream.